src/ProxyManager/ProxyGenerator/LazyLoadingGhostGenerator.php

```diff
@@ -134,8 +137,14 @@ static function (MethodGenerator $generatedMethod) use ($originalClass, $classGe
      *
      * @return MethodGenerator[]
      */
-    private function getAbstractProxiedMethods(ReflectionClass $originalClass): array
+    private function getAbstractProxiedMethods(ReflectionClass $originalClass, bool $skipDestructor): array
     {
+        $excludedMethods = ProxiedMethodsFilter::DEFAULT_EXCLUDED;
+
+        if ($skipDestructor) {
+            $excludedMethods[] = '__destruct';
```

Comment:
Nit: rename the option to "excludeDestructor" since that seems to be the internal wording.

Reply:
Not sure about this proposal: "exclude" here is used for a different thing.

Reply:
My thinking was that you're excluding the destructor from the list of proxied methods, but I defer to your (and @Ocramius') judgment on this 👍